src/Extensions/Diagnostics.HealthChecks/Internal/ServiceFabricHealthCheckPublisher.cs

```diff
@@ -12,33 +12,31 @@
 using Microsoft.Extensions.ObjectPool;
 using Microsoft.Omex.Extensions.Abstractions;
 using SfHealthInformation = System.Fabric.Health.HealthInformation;
-using SfHealthState = System.Fabric.Health.HealthState;
 
 namespace Microsoft.Omex.Extensions.Diagnostics.HealthChecks
 {
-	internal class ServiceFabricHealthCheckPublisher : IHealthCheckPublisher
+	internal class ServiceFabricHealthCheckPublisher : HealthCheckPublisher
 	{
-		internal const string HealthReportSourceId = nameof(ServiceFabricHealthCheckPublisher);
-
-		internal const string HealthReportSummaryProperty = "HealthReportSummary";
-
 		private readonly IAccessor<IServicePartition> m_partitionAccessor;
 
-		private readonly ILogger<ServiceFabricHealthCheckPublisher> m_logger;
+		private readonly IAccessor<StatelessServiceContext> m_serviceCtx;
 
-		private readonly ObjectPool<StringBuilder> m_stringBuilderPool;
+		private readonly ILogger<ServiceFabricHealthCheckPublisher> m_logger;
 
 		public ServiceFabricHealthCheckPublisher(
 			IAccessor<IServicePartition> partitionAccessor,
+			IAccessor<StatelessServiceContext> serviceCtx,
```

Comment:
This dependency should not be here, since then Stateful service won't be able to resolve publisher, and also I don't see how are you using it.